blueflood-http/src/main/java/com/rackspacecloud/blueflood/tracker/Tracker.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2013-2015 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.tracker;
+
+import com.rackspacecloud.blueflood.http.HTTPRequestWithDecodedQueryParams;
+import com.rackspacecloud.blueflood.io.Constants;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.management.JMX;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import java.lang.management.ManagementFactory;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class Tracker implements TrackerMBean {
+
+    private static final Logger log = LoggerFactory.getLogger(Tracker.class);
+    private Set tenantIds = new HashSet();
+
+    private static final String trackerName = String.format("com.rackspacecloud.blueflood.tracker:type=%s", Tracker.class.getSimpleName());
+    static MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+
+    public Tracker() {
+        registerMBean();
+    }
+
+    public void addTenant(String tenantId) {
+        if (!this.tenantIds.contains(tenantId)) {
```

Comment:
Actually you're right, HashSet should just ignore duplicate keys gracefully so I removed the check.